cosmoz-treenode-navigator.js

```diff
@@ -1,516 +1,249 @@
-		// stop event propagation to prevent iron-list trying to select the current item while it will be removed
-		e.preventDefault();
-		e.stopPropagation();
-	}
-	/**
-	 * Gets called if the selected node (path) has changed
-	 * @param {String} path - The path of the newly selected node
-	 * @return {undefined}
-	 */
-	_nodePathChanged(path) {
-		if (!path) {
-			this.highlightedNodePath = '';
-			return;
-		}
-		this.highlightedNodePath = path;
-	}
-	/**
-	 * Returns true, if the button should be visible
-	 * @param {Boolean} searching - If a search is currently executed
-	 * @param {String} openNodeLevelPath - The open node level
-	 * @return {Boolean} - The visibility of the button
-	 */
-	_showGlobalSearchBtn(searching, openNodeLevelPath) {
-		return searching && openNodeLevelPath !== '';
-	}
-	/**
-	 * Triggers a global search
-	 * @return {undefined}
-	 */
-	tryGlobalSearch() {
-		this._openNodePath = '';
-	}
-	/**
-	 * Returns true, if a search string is eligable to trigger a search
-	 * @param {String} value - The search string
-	 * @param {Number} searchMinLength - The minimum length of value to be valid
-	 * @return {Boolean} - If a search should be triggered
-	 */
-	_computeSearching(value, searchMinLength) {
-		return value && value.length >= searchMinLength && value !== '';
-	}
-	/**
-	 * Returns true, if the path of a node should be visible in the view
-	 * @param {Boolean} searching - If a search is currently executed
-	 * @param {Number} index - The node's current index in the list
-	 * @param {Array} dataPlane - The node list
-	 * @param {Object} parentSectionName - The nodes current parent section name
-	 * @return {Boolean} - If the path should be visible
-	 */
-	_renderSection(searching, index, dataPlane, parentSectionName) {
-		if (!searching || index == null || dataPlane == null || index >= dataPlane.length || parentSectionName == null) {
-			return false;
-		}
-		if (index === 0) {
-			return true;
-		}
-		const prevItem = dataPlane[index - 1];
-		if (prevItem.parentSectionName === parentSectionName) {
-			return false;
-		}
-		return true;
-	}
-	/**
-	 * Triggers a click event on the currentTarget
-	 * if space or enter key was pressed
-	 * @param {Event} e - The event
-	 * @return {undefined}
-	 */
-	_clickOnEnterOrSpace(e) {
-		if (e.keyCode === 13 || e.keyCode === 32) {
-			// enter or space pressed!
-			const fnName = 'click',
-				target = e.currentTarget,
-				// eslint-disable-next-line no-new-func
-				fn = new Function('target', 'fnName', 'return target.' + fnName + '()');
-			fn(target, fnName);
-		}
-	}
-	/**
-	 * Returns the classes of a row based its selection state
-	 * @param {String} classes - The default classes
-	 * @param {Boolean} selected - If the row is currently selected
-	 * @return {String} - The CSS classes
-	 */
-	_computeRowClass(classes, selected) {
-		return selected ? classes + ' selected' : classes;
-	}
-}
-customElements.define(CosmozTreenodeNavigator.is, CosmozTreenodeNavigator);
+		${ tree
```

Comment:
To avoid the ternary expressions  I'd suggest you use the `when` directive.
https://lit.dev/docs/templates/directives/#when

Reply:
Nice, thanks! Looks so much better now :)

Reply:
Woah! This directive alone makes upgrading to lit-html@2 worth it!